packages/hash/frontend/src/components/layout/MainContentWrapper.tsx

```diff
@@ -1,13 +1,75 @@
+import { IconButton, Fade, Box, Tooltip, styled } from "@mui/material";
 import { FunctionComponent } from "react";
+import { SIDEBAR_WIDTH } from "../../theme/components/navigation/MuiDrawerThemeOptions";
 
-import styles from "../../pages/index.module.scss";
+import { SidebarToggleIcon } from "../icons";
 import { PageSidebar } from "./PageSidebar/PageSidebar";
+import { useSidebarContext } from "./SidebarContext";
+
+const Main = styled("main", {
+  shouldForwardProp: (prop) => prop !== "sidebarOpen",
+})<{
+  sidebarOpen?: boolean;
+}>(({ theme, sidebarOpen }) => ({
+  flexGrow: 1,
+  padding: "60px 120px",
+  transition: theme.transitions.create("margin", {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  }),
+  marginLeft: `-${SIDEBAR_WIDTH}px`,
+  ...(sidebarOpen && {
+    transition: theme.transitions.create("margin", {
+      easing: theme.transitions.easing.easeOut,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    marginLeft: 0,
+  }),
+}));
 
 export const MainContentWrapper: FunctionComponent = ({ children }) => {
+  const { openSidebar, sidebarOpen } = useSidebarContext();
+
   return (
-    <div className={styles.MainWrapper}>
+    <Box
+      sx={{
+        display: "flex",
+        position: "relative",
+      }}
+    >
       <PageSidebar />
-      <main className={styles.MainContent}>{children}</main>
-    </div>
+      <Fade in={!sidebarOpen}>
+        <Tooltip title="Collapse Sidebar">
+          <IconButton
+            sx={{
+              height: 36,
+              width: 36,
+              borderRadius: "4px",
+              position: "absolute",
+              top: 32,
+              left: 32,
+              transform: "rotate(180deg)",
+
+              "&:hover": {
+                backgroundColor: ({ palette }) => palette.gray[10],
+                color: ({ palette }) => palette.gray[50],
+              },
+            }}
+            onClick={openSidebar}
+          >
+            <SidebarToggleIcon
+              sx={{
+                height: 20,
+                width: 20,
+              }}
+            />
+          </IconButton>
+        </Tooltip>
+      </Fade>
+      <Main sidebarOpen={sidebarOpen}>{children}</Main>
```

Comment:
I think this should now have a maximum height, so that you can't scroll past the sidebar:
<img width="1343" alt="image" src="https://user-images.githubusercontent.com/42802102/156781567-09e21e4e-2e3a-4ca5-abd2-5c9750e230ad.png">

When the content overflows, it should have a scrollbar.

Reply:
Thanks for catching, will update

Reply:
Awesome, thanks 👍